### PR TITLE
MINOR: Add tests for errors being thrown by NodeCG.listenFor()

### PR DESCRIPTION
**Pull request by madmatt**

### CODE REVIEW

File: test/api.js [hunk 1]
```diff
@@ -19,6 +19,21 @@ describe('client-side api', function () {
 				.call(done);
 		});
 
+		it('should ensure that duplicate bundleName-messageName pairs are ignored', done => {
+			e.apis.extension.listenFor('testMessageName', 'testBundleName', done);
+			e.apis.extension.listenFor('testMessageName', 'testBundleName', done);
```
Comment (Lange) [inline test/api.js]: Hm, I'm pretty sure this should actually throw an error. If it's not, then something is wrong.

Though, to be honest, the design of this API where it only accepts one listener is pretty silly. There's no reason for us to have this limitation, though me changing that would probably be considered a breaking change? Not sure, would have to think about that.

File: test/api.js [hunk 2]
```diff
@@ -19,6 +19,21 @@ describe('client-side api', function () {
...
+			done();
+		});
+
+		it('should produce an error if a callback isn\'t given', () => {
+			try {
+				e.apis.extension.listenFor('testMessageName', 'test');
+			} catch (err) {
+				assert.equal(err.message, 'argument "handler" must be a function, but you provided a(n) string');
+			}
+		});
```
Comment (Lange) [inline test/api.js]: Can you change this to use `assert.throws` instead? I find that to be a bit clearer. [See here](https://github.com/nodecg/nodecg/blob/master/test/schemas.js#L424-L432) for an example.

File: test/api.js (hunk 1 above)
Comment (madmatt) [reply test/api.js]: It does write an error (to the `Winston` error stream) but doesn't actually `throw` anything. So I ignore the error (couldn't figure out how to listen for error messages via Winston, or replace Winston with a mocked version), and just make sure that only one copy of the `testMessageName` listener gets registered.

Happy to remove this particular check if you think the code shouldn't work that way, I was just looking for some long-hanging fruit in the coverage reports to work on my unit testing knowledge in Node :)

Comment (Lange) [inline test/api.js]: Huh, wow yeah it just logs but doesn't `throw` lol. I've just made a commit that makes this actually `throw` now, can you rebase this branch and adjust the test to use `assert.throws`?

**Comment by madmatt** [comment]
Cool, I've rebased on `master` and updated the test now to reflect the new `throw` instead. Not sure if how I did it is the best way to declare an empty callback and then call it - I tried it another way and ran into issues where the test would timeout waiting for a call to `done();` - let me know if I can make it better :)

**Comment by Lange** [comment]
Thank you, merged!

**Pull request merged**